Add AKS with self-managed Calico docs (#2301)
tigera/docs: calico/getting-started/kubernetes/managed-public-cloud/aks-migrate.mdx

---
description: Switch AKS clusters between Azure-managed and self-managed Calico
title: Migrating AKS clusters between Azure-managed and self-managed Calico
---

> # Microsoft Azure Kubernetes Service (AKS)

Reviewer: Migrate from Azure-managed Calico to self-managed Calico

or similar

    - Your $[prodname] installation will be managed and supported by Azure.

1. Azure networking with self-managed $[prodname]

    - You will be able to make use of advanced $[prodname] features, like $[prodname] policy, Global Network policy, Whisker observability, etc
    - Your cluster will be supported by Azure, but the $[prodname] features would not be supported by Azure. You will need to upgrade $[prodname] yourself. For community support, get an invite to our slack channel at: http://slack.project$[prodname].org/ or raise an issue in GitHub. Commercial support plans are also available.

1. use $[prodname] for Networking and Policy in AKS, in place of the default Azure VPC networking

    - This allows you to take advantage of the full set of $[prodname] networking features.
    - Your cluster will be supported by Azure, but the $[prodname] features would not be supported by Azure. You will need to upgrade $[prodname] yourself. For community support, get an invite to our slack channel at: http://slack.project$[prodname].org/ or raise an issue in GitHub. Commercial support plans are also available.

> It is possible to migrate an existing AKS cluster between the first two modes. The following sections describe how to do this.

Reviewer: This page describes how to migrate your AKS cluster from using Azure-managed Calico for network policy to using self-managed Calico for network policy. Doing this gives you access to the all of Calico's network policy features, including global network policy, staged network policy, and policy tiers.

(Let's keep the short description brief)

## Value

AKS has built-in support for $[prodname], providing a robust implementation of the full Kubernetes Network Policy API.

> AKS users wanting to go beyond Kubernetes network policy capabilities can install a self-managed $[prodname] to make full use of advanced features like the $[prodname] Network Policy API, Whisker for Observability, etc.

Reviewer: If you want to expand on the differences between the two policy implementations, here's the spot.

Perhaps something like this:

```
## Understanding Azure- and self-managed Calico for policy on AKS

### Azure-managed Calico on AKS

Azure offers Calico as a managed option for network policy enforcement.
When you create an AKS cluster with `--network-policy calico`, Azure handles the installation and lifecycle of Calico's policy enforcement components.
This managed version provides a robust, supported implementation of the standard Kubernetes Network Policy API.
This is for users who need to enforce basic pod-to-pod communication rules, such as ingress and egress traffic based on pod labels and namespaces.
However, this managed service is limited to the features within that standard API.
You cannot change the Calico version, and you don't get access to Calico-specific custom resources or advanced features.

## Self-managed Calico on AKS

To access all of Calico's policy features, you must install it yourself on an AKS cluster.
You can do this on an existing AKS cluster by changing the network policy configuration (--network-policy none) and then deploying Calico using the Tigera Operator.
This approach gives you access to the Calico Network Policy API, which offers a richer set of features beyond the standard Kubernetes policies, including:
* Global network policies: Rules that apply to all pods and endpoints across all namespaces, which is not possible with the standard Kubernetes Network Policy.
* Order and priority: The ability to order policies and apply explicit deny rules, giving you more granular control over traffic flow.
* Extended endpoint types: Calico policies can be applied to endpoints beyond just pods, such as host interfaces and VMs.

The key distinction is that the managed Azure version gives you a simple, hands-off experience for basic policy, while a self-managed installation gives you complete control and access to all of Calico's advanced network security features.

## Prerequisites

Use this procedure if you have an existing cluster which uses Azure CNI and Azure-managed $[prodname] for policy, and you now wish to migrate to self-managed $[prodname] in order to use Advanced $[prodname] features like $[prodname] Network Policy or Whisker Observability.

> You can use this command to check your existing cluster configuration.  If you get the output shown below, then you have Azure CNI and Azure-managed $[prodname] for policy and can use this procedure to migrate to self-managed $[prodname]:

Reviewer: * You have an AKS cluster that uses Azure-managed Calico for network policy.
  To confirm that your cluster has this configuration, run the following command:
  ```bash
  command
  ```
  ```bash title='Expected output'
  output
  ```
  
<img width="567" height="173" alt="image" src="https://github.com/user-attachments/assets/722fd619-3bff-4eae-a2cb-665023040820" />

Reviewer: For simplicity.

  ```bash
  az aks show --name my-calico-cluster --resource-group my-calico-rg  \
    --query "networkProfile.networkPlugin, networkProfile.networkPolicy"
[
  "azure",
  "calico"
]
  ```

:::warning
Application connectivity may be disrupted during the migration

Network policy will be removed during the migration and must be re-applied after the migration.
:::

> 

Reviewer: `## Repeat the title of the procedure.`

Write a few lines to describe what's happening so people can follow along more easily. Something like:

During this migration, you will save your current network policies, remove the Azure-managed Calico, install a self-managed Calico, and then reapply your network policies.

> 1. Get all existing network policies from all namespaces and write them out to a file

Reviewer: ```suggestion
1. Get all existing network policies from all namespaces and write them out to a file:
```

    ```bash

>     kubectl get netpol -A -o yaml  > network-policy.yaml

Reviewer: ```suggestion
    kubectl get netpol -A -o yaml > network-policy.yaml
```

    ```

>     Review the file to check everything you expect to be there, is.

Reviewer: ```suggestion
    Review the file to make sure it contains all the policies you expect to be there.
```

> 1. Do the migration

Reviewer: I think this should be more like:
Remove the Azure-managed Calico by modifying the AKS cluster configuration:

    ```bash
    az aks update --name my-calico-cluster --resource-group my-calico-rg --network-policy=none

>     ```

Reviewer: Can we format these like you did in the install doc?

Author: Can you expand on that?  Do you mean multiline, with one flag per line?

> 1. Confirm that the migration has completed successfully

Reviewer: ```suggestion
1. Verify the new configuration:
```

    ```bash
    az aks show --name my-calico-cluster --resource-group my-calico-rg  \
      --query "networkProfile.networkPlugin, networkProfile.networkPolicy"
    [
    "azure",
    "none"
    ]

>     ```

Reviewer: Change as above with 'expected output'

1. Install $[prodname].

>     Note that we do not install CRDs here, since they may already exist. Operator will check and create any missing ones.

Reviewer: Why not leave the CRD line in? What's the harm?

Author: If they exist, the command will fail, and I'd rather not have to deal with that :)

Reviewer: Great, that's fine then. I'd format this as an actual note, and put it at the end of of this item, after the code block.

    ```bash
    kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.30.3/manifests/tigera-operator.yaml
    ```

> 1. Configure the $[prodname] installation

Reviewer: Change as in installation doc

Author: Again, "configure the Calico installation" is literally what this section is doing.  Lets talk about this

    # For more information, see: https://docs.tigera.io/calico/latest/reference/installation/api#operator.tigera.io/v1.APIServer
    apiVersion: operator.tigera.io/v1
    kind: APIServer
    metadata:
      name: default
    spec: {}

    ---

    # This optional section enables the Calico Goldmane flow aggregator.
    apiVersion: operator.tigera.io/v1
    kind: Goldmane
    metadata:
      name: default

    ---

    # This optional section enables the Calico Whisker observability UI.
    apiVersion: operator.tigera.io/v1
    kind: Whisker
    metadata:
      name: default
    EOF
    ```

1. Confirm that $[prodname] is up and running
    ```bash
    kubectl get tigerastatus

    NAME        AVAILABLE   PROGRESSING   DEGRADED   SINCE
    apiserver   True        False         False      100s
    calico      True        False         False      50s
    goldmane    True        False         False      55s
    ippools     True        False         False      105s
    whisker     True        False         False      80s
    ```
Check that all components report `AVAILABLE` as `True`.

1. Re-apply all existing policies
    ```bash
    kubectl apply -f network-policy.yaml

>     ```

Reviewer: Is there a good verification that this is right?

Author: Not really.  If it doesn't succeed, there will be an error?

Reviewer: Fair enough. I wondered whether it would be worth the bother to run the output command again to compare to what you put in? But that may enter levels of silliness we don't care for.

Author: I guess you could run `kubectl get networkpolicy -A -o yaml` and eyeball the output?

Reviewer: Nah let's leave it for now.

---
description: Switch AKS clusters between Azure-managed and self-managed Calico

> title: Migrating AKS clusters between Azure-managed and self-managed Calico

Reviewer: ```suggestion
title: Migrating AKS clusters to self-managed Calico
```
Shorten this so it fits in the sidebars. Currently truncated.

    spec:
      kubernetesProvider: AKS
      cni:
        type: AzureVNET
    ---

    # This optional section configures the Calico API server.
    # For more information, see: https://docs.tigera.io/calico/latest/reference/installation/api#operator.tigera.io/v1.APIServer
    apiVersion: operator.tigera.io/v1
    kind: APIServer
    metadata:
      name: default
    spec: {}

    ---

    # This optional section enables the Calico Goldmane flow aggregator.
    apiVersion: operator.tigera.io/v1
    kind: Goldmane
    metadata:
      name: default

    ---

    # This optional section enables the Calico Whisker observability UI.
    apiVersion: operator.tigera.io/v1
    kind: Whisker
    metadata:
      name: default
    EOF
    ```

1. Confirm that $[prodname] is up and running
    ```bash
    kubectl get tigerastatus

    NAME        AVAILABLE   PROGRESSING   DEGRADED   SINCE
    apiserver   True        False         False      100s
    calico      True        False         False      50s
    goldmane    True        False         False      55s
    ippools     True        False         False      105s
    whisker     True        False         False      80s
    ```

> Check that all components report `AVAILABLE` as `True`.

Reviewer: Move in line with above step.

Reviewer: Occurs at least twice in the PR.

---
description: Switch AKS clusters between Azure-managed and self-managed Calico
title: Migrating AKS clusters between Azure-managed and self-managed Calico
---

# Migrate from Azure-managed Calico to self-managed Calico

> ## Big picture

Reviewer: rm
